indexing-service/src/main/java/org/apache/druid/indexing/overlord/ForkingTaskRunner.java

```diff
@@ -327,6 +327,13 @@ public TaskStatus call()
                           command.add(nodeType);
                         }
 
+                        // If the task type is queryable, we need to load broadcast segments on the peon, used for
+                        // join queries
+                        if (task.supportsQueries()) {
+                          command.add("--loadBroadcastSegments");
```

Comment:
Hmm, the CI failure doesn't like missing tests for this change. Thinking about it for a bit, what will happen if the peon always loads the stuffs for loading broadcast segments? If I'm reading the code correctly, the coordinator will not broadcast segments to batch tasks no matter whether they are running on neither middleManagers nor indexers. [The realtime tasks announce themselves when they run on middleManagers or don't announce when they run on indexers](https://github.com/apache/druid/blob/master/indexing-service/src/main/java/org/apache/druid/indexing/seekablestream/SeekableStreamIndexTaskRunner.java#L420-L423), but the batch tasks never announce themselves. So, it seems like, if the peon always loaded the stuffs for broadcast segment loading, those stuffs could be loaded for batch tasks as well but wouldn't do anything? If that's the case, I'm fine with always loading them rather than adding this new parameter and tests for it.

Reply:
Hm, I saw the batch tasks loading things from segment cache, which I was trying to avoid right now since they could be loaded on-heap for indexed tables, for performance/memory load reasons.

For the coverage, I'm thinking we should ignore the coverage failure on ForkingTaskRunner and CliPeon for this PR, does that sound okay to you?

Reply:
Ah, the segment cache is a good point. It makes sense.

> For the coverage, I'm thinking we should ignore the coverage failure on ForkingTaskRunner and CliPeon for this PR, does that sound okay to you?

I think it's fine.